### PR TITLE
core: pager can use memory between SRAM start and core load address

### PR DESCRIPTION
**Pull request by etienne-lms**
If core is loaded some 4kB pages above the start of the physical internal ram, some 4kB memory block will not be used by the pager.

This situation can occur if the beginning of the internal ram is used by a bootloader. Bootloader must load op-tee above its own used memory. Such bootloader memory is freely available to op-tee core (pager).

This change adds the physical memory between TEE RAM base address and the op-tee entry point address to the pager page pool. This change also default maps this area so that pager identifies physical pages as valid page addresses.

I found this issue while verifying support for various locations of a flat [mapped coherent memory](https://github.com/OP-TEE/optee_os/pull/1533).

### CODE REVIEW

File: core/arch/arm/kernel/generic_boot.c [hunk 1]
```diff
@@ -299,6 +297,10 @@ static void init_runtime(unsigned long pageable_part)
 	tee_pager_add_pages((vaddr_t)__pageable_start + init_size,
 			(pageable_size - init_size) / SMALL_PAGE_SIZE, true);
 
+	/* Pager can also use pages before unpaged memory if any */
```
Comment (jenswi-linaro) [inline core/arch/arm/kernel/generic_boot.c]: It could be worth commenting on when and why there may be pages before unpaged memory.

**Comment by jforissier** [comment]
@etienne-lms can you please rebase and address the conflicts with master?

**Comment by etienne-lms** [comment]
ok

**Comment by jforissier** [comment]
@etienne-lms I think you could have just applied Jens' R-b, no? Did you change something significant that he did not acknowledge?

**Comment by etienne-lms** [comment]
ok, i apply and squash.

**Comment by jforissier** [comment]
Thanks!

**Pull request merged**